### PR TITLE
Remove tabs from cart and checkout

### LINKED ISSUE #4706: [mobile ux] edit cart displaying on home tab + shop tab; check out displays snail
## Description
Edit cart screen displays the home tab content, and then seems to break shop tab.

## Expected Behavior
When I click 'Edit cart' I am taken to the list of products in my cart (in which tab @yukoosawa does the edit cart detail go?), and I can then click on 'shop' tab which takes me back to the product list, with no cart product list displaying

## Actual Behaviour
When I click 'Edit cart' I am taken to the home tab, with the content above and then the list of products in my cart displays underneath. 
Screen Shot 2020-01-22 at 5 57 35 pm

I then click on 'shop tab' and I am taken to the shop tab but it thinks I've done a search, displays a weird message, shows the loading icon continually, and also displays the cart underneath it all. (note: if I edit the cart here and then hit update I am taken back to the home tab with the updated cart displaying underneath copy.)
Screen Shot 2020-01-22 at 5 58 01 pm
Screen Shot 2020-01-22 at 6 06 57 pm
Screen Shot 2020-01-22 at 6 07 04 pm

Then I try to check out - snail fail :(

## Steps to Reproduce
1. Add a product to your cart
2. Scroll to end of product list, click on 'Edit cart'
3. Home tab opens, list of products display underneath the shopfront copy
4. Click on shop tab
5. Shop tab opens, zero search results + loading icon + error message + variables not processing displays with edit cart list.
6. Update an item in the cart, then hit update
7. Home tab opens, list of products that has been updated displays underneath the shopfront copy.
8. Open 

## Animated Gif/Screenshot
https://www.loom.com/share/4565ba63167b4e938bfde078341f7cbe

## Workaround
No workaround; checkout broke.

## Your Environment

* Version used: UK staging
* Browser name and version: Chrome Version 79.0.3945.117 (Official Build) (64-bit)
* Operating System and version (desktop or mobile): Desktop Apple 10.14.6

### PR DESCRIPTION
**Pull request by Matt-Yorkley**
#### What? Why?

Closes #4706

Splits up the `shopping_shared/details` partial into `shopping_shared_header` and `shopping_shared/tabs`, and includes the tabs only in the shop page (not in cart and checkout).

#### What should we test?

Cart and checkout pages are working ok (not displaying shop page content).

#### Release notes

Removed shop content tabs from cart and checkout.

Changelog Category: Changed

### CODE REVIEW

**Comment by Matt-Yorkley** [comment]
As mentioned in the connected issue, this is a quick PR to resolve the `S1`, but it needs some product feedback. @daniellemoorhead @RachL

**Comment by Matt-Yorkley** [comment]
Additional explanation: we used to show all the shop tabs on the cart and checkout pages, which defaulted to being closed. The shop tabs have changed a lot in a recent PR; tab selection logic has changed, the tabs are always open, the product list is included in the tabs etc etc.

It's not working nicely when we redisplay all that content in the cart and checkout. This PR removes the shop tabs and content from the cart and checkout pages (as the quickest and simplest solution).

BEFORE:
Screenshot from 2020-01-22 11-52-36

AFTER:
Screenshot from 2020-01-22 11-49-19

**Comment by lin-d-hop** [comment]
Personally I think this new logic is fine. 

It would be nice to have matamo data to know how often these tabs are used in these pages. In absence my common sense and intuition say that the information in tabs is not a priority in Cart and Checkout.

Happy to be overruled if others disagree.... but let's not hold up the next release for this!

**Comment by Matt-Yorkley** [comment]
Yeah I'm mostly thinking about tomorrow's release here. The alternate option is that we revert the recent mobile improvements in `master` before the release, then resolve this and the merging of other related PRs by next week.

File: app/views/shopping_shared/_tabs.html.haml [hunk 1]
```diff
@@ -1,11 +1,13 @@
-- shop_tabs.each do |tab|
-  = render "shopping_shared/tabs/#{tab[:name]}"
+- if (@order.andand.distributor || current_distributor) == current_distributor
```
Comment (luisramos0) [inline app/views/shopping_shared/_tabs.html.haml]: why is this needed? if I just switched shops I still want to see the tabs for the new shop, right?

Comment (mkllnk) [reply app/views/shopping_shared/_tabs.html.haml]: If you switched shops then `current_distributor` should be the new shop.

This is a quick fix for another bug. I can't find the issue right now but if your current distributor is ShopA and then you look at an order receipt for ShopB then the tabs use current distributor and looking up contact details shows you ShopA. So this `if` clause is not fixing the problem but avoiding it by hiding the tabs in that case. Better than before, I think.

**Comment by daniellemoorhead** [comment]
New recording of the process:
https://www.loom.com/share/146ae8d4596c453b80b235265e22958d

Notes:
1. Edit cart shows up correctly and nothing is broken when I use it. The lack of tabs feels like a gap, on desktop the user can click on the shop name in the menu or on any device can click the 'Continue shopping' button but we definitely need to bring the tabs back.

2. I still get a snail fail on checkout. This is not fixed @Matt-Yorkley @mklink.

**Comment by daniellemoorhead** [comment]
OK, so I tested on Mac/Chrome + on Browserstack Windows 10 / Chrome + IE11. The edit cart function works as in the video above, and I can check out. So yep, the checkout fail is a UK staging thing.

This is now ready to go. I'll add another story to get the tabs added back on to this screen.

File: app/views/shopping_shared/_tabs.html.haml (hunk 1 above)
Comment (Matt-Yorkley) [reply app/views/shopping_shared/_tabs.html.haml]: This bit of logic was present in the original `details.html` partial, I just moved it here when splitting the file into `header.html` and `tabs.html`.

https://github.com/openfoodfoundation/openfoodnetwork/pull/4707/files#diff-d2ee7c7d99b77ac7210e1d930482692eR1
https://github.com/openfoodfoundation/openfoodnetwork/pull/4707/files#diff-d2ee7c7d99b77ac7210e1d930482692eL18

Comment (luisramos0) [reply app/views/shopping_shared/_tabs.html.haml]: ah, makes sense! Thanks for explaning Maikel. I think this "looking at an order receipt for ShopB" deserves a comment in the code.

**Comment by luisramos0** [comment]
awesome work people!!! Merging.
